### DISCUSSION TITLE
Moving Periodic Surfaces conformally during optimization.

### DISCUSSION THREAD

**Discussion by NAnand-TUD** [su2code/SU2#2000] [Q&A]
Hello SU2ers,

I am trying to solve an optimization problem in which the periodic surface boundary should move conformally. Is there any quick way to do it? 

One way is to write an algorithm to handle this in CVolumetricMovement::SetBoundaryDisplacements. Here the PERIODIC_BC flag needs to updated and some adaption needs to be made. Before I commit to that I would like to hear if there is a better way to handle this issue.

Additionally, I see other is MARKER_DEFORM_MESH_SYM_PLANE as an option in config.cpp but it is not being used in the source code. Is that the case or am I missing something here? 

Thanks,
Nitish

**Comment by pcarruscag** [comment]
That option is only used by the new mesh solver for now.
The "best" way to handle this would be to make the deformation solver aware of periodicit, but that is going to be challenging.
Can you apply a deformation to both periodic boundaries as a function of the design deformation?

**Reply by NAnand-TUD** [reply]
![image](https://user-images.githubusercontent.com/22835014/229709379-28ecd484-f97e-4dec-b7be-69f6aaf695ed.png)

So my geometry is like this, where I specify the deformation on the surface (quarter cylinders) and I want periodic boundaries to move conformally in 3D.  Here the surface movement of the two cylinders is conformal.

**Reply by pcarruscag** [reply]
You can define those patches as symmetry for su2_def and _dot. If the deformation is conformal the result should also be fine.
Alternatively you can expand the ffd boxes to include part those patches in the design surfaces. Then you need to move inner points and fix the outer point to ensure continuity.

**Reply by NAnand-TUD** [reply]
![image](https://user-images.githubusercontent.com/22835014/229737010-ec7bc2bd-cc0b-4e42-9cdf-c2bc5baf23e7.png)
Periodic1
![image](https://user-images.githubusercontent.com/22835014/229737129-13bb7010-3b34-40eb-b2be-eeb04d28a869.png)
Periodic 2

The figure above is when I move the surfaces by marking them as SYM_BCs. In the figures, you can see that the deformation of the two periodic surfaces is different on either side, although conformal deformation was imposed. If I treat them as PER_BCs the surfaces come out-of-plane as expected.

FFD could have been an option but we are sticking to CAD parametrization so far.

Maybe we can think a bit about this issue during DevMeeting.
